app/src/main/java/tech/relaycorp/gateway/background/endpoint/EndpointPreRegistrationService.kt

```diff
@@ -0,0 +1,55 @@
+package tech.relaycorp.gateway.background.endpoint
+
+import android.app.Service
+import android.content.Intent
+import android.os.Handler
+import android.os.IBinder
+import android.os.Looper
+import android.os.Message
+import android.os.Messenger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.runBlocking
+import tech.relaycorp.gateway.background.component
+import tech.relaycorp.gateway.domain.endpoint.EndpointRegistration
+import javax.inject.Inject
+
+class EndpointPreRegistrationService : Service() {
+    private val scope get() = CoroutineScope(Dispatchers.IO)
```

Comment:
Is this the right scope to use? I wasn't sure whether to add a `SupervisorJob` but I didn't think it'd be necessary here from what I read.